pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -193,6 +193,9 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
 	processNode := func(i int) {
 		nodeInfo := nodeNameToInfo[allNodeNames[i]]
 		if nodeInfo.Node() != nil {
+			if !nodeInfilteredNodes(nodeInfo, nodes) {
```

Comment:
Although it's correct in semantics, this method can be costy. Could you 

1. change the code like this:

    ```diff
    diff --git a/pkg/scheduler/algorithm/priorities/interpod_affinity.go     b/pkg/scheduler/algorithm/priorities/interpod_affinity.go
    index 32cf27c83b..55c7104402 100644
    --- a/pkg/scheduler/algorithm/priorities/interpod_affinity.go
    +++ b/pkg/scheduler/algorithm/priorities/interpod_affinity.go
    @@ -121,9 +121,9 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
     	hasAffinityConstraints := affinity != nil && affinity.PodAffinity != nil
     	hasAntiAffinityConstraints := affinity != nil && affinity.PodAntiAffinity != nil
     
    -	allNodeNames := make([]string, 0, len(nodeNameToInfo))
    -	for name := range nodeNameToInfo {
    -		allNodeNames = append(allNodeNames, name)
    +	filteredNodeNames := make([]string, 0, len(nodes))
    +	for _, node := range nodes {
    +		filteredNodeNames = append(filteredNodeNames, node.Name)
     	}
     
     	// convert the topology key based weights to the node name based weights
    @@ -191,7 +191,7 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
     		return nil
     	}
     	processNode := func(i int) {
    -		nodeInfo := nodeNameToInfo[allNodeNames[i]]
    +		nodeInfo := nodeNameToInfo[filteredNodeNames[i]]
     		if nodeInfo.Node() != nil {
     			if hasAffinityConstraints || hasAntiAffinityConstraints {
     				// We need to process all the nodes.
    @@ -211,7 +211,7 @@ func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, node
     			}
     		}
     	}
    -	workqueue.ParallelizeUntil(context.TODO(), 16, len(allNodeNames), processNode)
    +	workqueue.ParallelizeUntil(context.TODO(), 16, len(filteredNodeNames), processNode)
     	if pm.firstError != nil {
     		return nil, pm.firstError
     	}
    ```

1. and add an unit test?

Thanks for the catch!

Reply:
The unit test could be like:
- create 4 nodes
- prepare a pod with
  - node affinity with **_hard_** requirement (requiredDuringSchedulingIgnoredDuringExecution) to node1 and node2, and also 
  - and pod affinity with **_soft_** requirement (preferredDuringSchedulingIgnoredDuringExecution)

So that verify the priority runs only on node1 and node2.